File: JAVA/L3/hexagon.java
public class hexagon extends shape{
	double edge;

	hexagon(){
		edge = 0;
	}
	hexagon(double edge){
		this.edge = edge;
	}

	public double getArea(){
		return (edge * edge * 3 * Math.sqrt(3))/2;
	}
	public double getPerimeter(){
		return 6 * edge;
	}
	@Override
	public String toString(){
		return "hexagon with edge: " + edge;
	}
}

File: CPP/L3/Hexagon.cpp
#include "Hexagon.h"

Hexagon::Hexagon(double e): edge(e) {};
Hexagon::Hexagon(char* e){
	edge = atof(e);
	if (edge <= 0) throw "Hexagon edge argument must have a positive value";
}

double Hexagon::getArea() const {
	return 3*sqrt(3)*edge*edge/2;
}
double Hexagon::getPerimeter() const {
	return 6*edge;
}
void Hexagon::print(){
	std::cout << "Hexagon Area: " << getArea() << " Perimeter: " << getPerimeter() << "\n";
}

